### ISSUE TITLE
Protocol checking against `typing_extenstion.Protocol` fails

### ISSUE THREAD

**Issue by iliapolo** [agronholm/typeguard#479]
### Things to check first

- [X] I have searched the existing issues and didn't find my bug already reported there

- [X] I have checked that my bug is still present in the latest release

### Typeguard version

4.3.0

### Python version

Python 3.10.4

### What happened?

Expected typeguard to ignore private attributes of the `typing_extension.Protocol` so that type checking would pass.

> Note: Honestly this was my first intuition, but I realize it might be out of scope as I'm not sure what your stance is with respect to supporting the `typing_extensions` package. Perhaps this issue is more a feature request to allow configuring custom attributes that typeguard will ignore when typechecking protocols.

### How can we reproduce the bug?

```python
import typing_extensions
import builtins
import typeguard

class IDo(typing_extensions.Protocol):
    def do_something(self) -> builtins.str:
        ...

def doer(do: IDo) -> builtins.str:
    typeguard.check_type(value=do, expected_type=IDo)
    return do.do_something()

class DefaultDo:
    def do_something(self) -> builtins.str:
        return "did something"
    
print(doer(DefaultDo()))
```

Running the above program will result in:

```console
typeguard.TypeCheckError: __main__.DefaultDo is not compatible with the IDo protocol because it has no attribute named '__protocol_attrs__'
```

Indeed, extending the `typing_extensions.Protocol` [adds the `__protocol_attrs__`](https://github.com/python/typing_extensions/blob/main/src/typing_extensions.py#L602) attribute and causes the mismatch. I did notice however that typeguard has provisions to ignore private attributes of the `typing.Protocol` class:

 https://github.com/agronholm/typeguard/blob/016f8139f5a0a63147d68df9558cc5584cd2c49a/src/typeguard/_checkers.py#L668-L671

So I was thinking this might be extended to support `typing_extension.Protocol`? If not, it would be great if we had some way to inject custom attributes we'd like typeguard to ignore.

**Comment by agronholm** [comment]
Does Mypy pass your code in strict mode?

**Comment by iliapolo** [comment]
Looks like it:

```console
❯ mypy --version                                                                                                                                                                                                                                    [12:51:41]
mypy 1.11.2 (compiled: yes)

❯ mypy --strict tests/exp.py                                                                                                                                                                                                                        [12:51:45]
Success: no issues found in 1 source file
```

**Comment by agronholm** [comment]
Yeah, I recall something like this was reported before. I'll get it fixed for the next release, but I'm currently busy trying to make new releases in two other projects. I'll tackle typeguard after that.

**Comment by iliapolo** [comment]
@agronholm If we agree on an approach here i'm happy to contribute a PR. Would the solution be using `dir(typing_extension.Protocol)` instead of `dir(typing.Protocol)`? maybe a union of both?

**Comment by thetorpedodog** [comment]
> What this suggests is that you could use the attributes of an empty protocol (like GenericEmpty above) to generate the list of things you need to ignore when checking a protocol.

Actually, upon further research, the solution might be simpler: use `typing_extensions.get_protocol_members` (to become `typing.get_protocol_members` in 3.13). If you don’t want to have a hard dependency on `typing_extensions`, this might require you to vendor [the relevant functions](https://github.com/python/typing_extensions/blob/28493326b1a0f10f46e9a1ed85ff8da6341942ca/src/typing_extensions.py#L3569-L3608).

**Comment by agronholm** [comment]
> Actually, upon further research, the solution might be simpler: use `typing_extensions.get_protocol_members` (to become `typing.get_protocol_members` in 3.13). If you don’t want to have a hard dependency on `typing_extensions`, this might require you to vendor [the relevant functions](https://github.com/python/typing_extensions/blob/28493326b1a0f10f46e9a1ed85ff8da6341942ca/src/typing_extensions.py#L3569-L3608).

Typeguard already has a hard dependency on `typing_extensions`, so I'll just use that function then. Thanks!

**Comment by agronholm** [comment]
This will be fixed with https://github.com/agronholm/typeguard/pull/490.

**Comment by agronholm** [comment]
This was fixed in v4.4.0.
